Failure to report build error

(( Ps: Thanks for all the recent improvements to CDash :smile: ))

The real build error is:

```
[  2%] Building CXX object QuickTCGA/Logic/NucleusSeg_Yi/CMakeFiles/NucleiSegLib.dir/Normalization.cpp.o
/Users/kitware/Dashboards/Nightly/S-0-E-b/SlicerPathology/QuickTCGA/Logic/NucleusSeg_Yi/Normalization.cpp:221:24: error: call to 'pow' is
      ambiguous
        LMS_ptr[j*3] = pow(10.0, log_LMS_ptr[j*3]);
                       ^~~
/Developer/SDKs/MacOSX10.6.sdk/usr/include/architecture/i386/math.h:343:15: note: candidate function
extern double pow ( double, double );
              ^
/Developer/SDKs/MacOSX10.6.sdk/usr/include/c++/4.2.1/cmath:356:3: note: candidate function
  pow(float __x, float __y)

[...]

/Developer/SDKs/MacOSX10.6.sdk/usr/include/c++/4.2.1/cmath:372:3: note: candidate function
  pow(long double __x, int __n)
  ^
3 errors generated.
make[2]: *** [QuickTCGA/Logic/NucleusSeg_Yi/CMakeFiles/NucleiSegLib.dir/Normalization.cpp.o] Error 1
make[1]: *** [QuickTCGA/Logic/NucleusSeg_Yi/CMakeFiles/NucleiSegLib.dir/all] Error 2
make: *** [all] Error 2
```

Looking at the `Build.xml`, the error is captured:

```
[...]
               <Error>
                        <BuildLogLine>10</BuildLogLine>
                        <Text>/.../SlicerPathology/QuickTCGA/Logic/NucleusSeg_Yi/Normalization.cpp:221:24: error: call to 'pow' is ambiguous</Text>
                        <SourceFile>QuickTCGA/Logic/NucleusSeg_Yi/Normalization.cpp</SourceFile>
                        <SourceLineNumber>221</SourceLineNumber>
                        <PreContext>Scanning dependencies of target ConfigureAdditionalLauncherSettings
[  1%] Configuring: AdditionalLauncherSettings.ini
[  1%] Built target ConfigureAdditionalLauncherSettings
Scanning dependencies of target NucleiSegLib
[  1%] Building CXX object QuickTCGA/Logic/NucleusSeg_Yi/CMakeFiles/NucleiSegLib.dir/BinaryMaskAnalysisFilter.cxx.o
Scanning dependencies of target CopyvtkSlicerQuickTCGAModuleLogicPythonScriptFiles
[  2%] Copying python Script: vtkSlicerQuickTCGAModuleLogic.py
[  2%] Built target CopyvtkSlicerQuickTCGAModuleLogicPythonScriptFiles
[  3%] Building CXX object QuickTCGA/Logic/NucleusSeg_Yi/CMakeFiles/NucleiSegLib.dir/Normalization.cpp.o
</PreContext>
                        <PostContext>        LMS_ptr[j*3] = pow(10.0, log_LMS_ptr[j*3]);
                       ^~~
</PostContext>
                        <RepeatCount>0</RepeatCount>
                </Error>

[...]

                <Error>
                        <BuildLogLine>31</BuildLogLine>
                        <Text>/.../SlicerPathology/QuickTCGA/Logic/NucleusSeg_Yi/Normalization.cpp:222:26: error: call to 'pow' is ambiguous</Text>
                        <SourceFile>QuickTCGA/Logic/NucleusSeg_Yi/Normalization.cpp</SourceFile>
                        <SourceLineNumber>222</SourceLineNumber>
                        <PreContext></PreContext>
                        <PostContext>        LMS_ptr[j*3+1] = pow(10.0, log_LMS_ptr[j*3+1]);
                         ^~~
</PostContext>
                        <RepeatCount>0</RepeatCount>
                </Error>

[...]
```


but CDash does not report the string `call to 'pow' is ambiguous`:

Looks like only the `Postcontext` is listed with omission of the `Text` ? 

![2016 05 02_cdash_error-not-fully-reported](https://cloud.githubusercontent.com/assets/219043/14957201/0534f770-1052-11e6-9118-fcd1f23bd27b.png)

--- a/models/builderror.php
+++ b/models/builderror.php
@@ -124,7 +124,7 @@ class BuildError
 
         if (isset($data['subprojectid'])) {
             $marshaled['subprojectid'] = $data['subprojectid'];
-            $marshaled['subprojectname'] = $data['name'];
+            $marshaled['subprojectname'] = $data['subprojectname'];
         }
 
         return $marshaled;

--- a/public/js/controllers/viewBuildError.js
+++ b/public/js/controllers/viewBuildError.js
@@ -6,11 +6,25 @@ CDash.controller('BuildErrorController',
       method: 'GET',
       params: $rootScope.queryString
     }).success(function(cdash) {
-      for (var i in cdash.errors) {
-        // Handle the fact that we add HTML links to compiler output.
-        cdash.errors[i].precontext = $sce.trustAsHtml(cdash.errors[i].precontext);
-        cdash.errors[i].postcontext = $sce.trustAsHtml(cdash.errors[i].postcontext);
-        cdash.errors[i].text = $sce.trustAsHtml(cdash.errors[i].text);
+      // Handle the fact that we add HTML links to compiler output.
+      var trustErrorHtml = function (error) {
+          error.precontext = $sce.trustAsHtml(error.precontext);
+          error.postcontext = $sce.trustAsHtml(error.postcontext);
+          error.text = $sce.trustAsHtml(error.text);
+          return error;
+      };
+
+      // Errors are either cdash.errors, or all values in cdash.errors.*
+      if (Array.isArray(cdash.errors)) {
+          for (var i in cdash.errors) {
+              cdash.errors[i] = trustErrorHtml(cdash.errors[i]);
+          }
+      } else {
+          for (var subproject in cdash.errors) {
+              for (var error in cdash.errors[subproject]) {
+                  cdash.errors[subproject][error] = trustErrorHtml(cdash.errors[subproject][error]);
+              }
+          }
       }
       renderTimer.initialRender($scope, cdash);
 

--- a/tests/js/e2e_tests/viewBuildError.js
+++ b/tests/js/e2e_tests/viewBuildError.js
@@ -19,4 +19,15 @@ describe("viewBuildError", function() {
     browser.get('viewBuildError.php?buildid=6&type=1');
     expect(browser.getPageSource()).toContain("10</b> Warnings");
   });
+
+  it("displays build errors inline", function() {
+    browser.get('viewBuildError.php?buildid=67&type=0');
+    expect(browser.getPageSource()).toContain("error: 'foo' was not declared in this scope");
+  });
+
+  it("displays build errors inline on parent builds", function() {
+    browser.get('viewBuildError.php?buildid=66&type=0');
+    expect(browser.getPageSource()).toContain("some-test-subproject");
+    expect(browser.getPageSource()).toContain("error: 'foo' was not declared in this scope");
+  });
 });

